api/controllers/blocks.js

```diff
@@ -87,4 +105,104 @@ BlocksController.getBlocks = function(context, next) {
 	});
 };
 
+/**
+ * Get filtered list of blocks (without transactions - BasicBlock).
+ *
+ * @private
+ * @func _list
+ * @param {Object} filter - Conditions to filter with
+ * @param {string} filter.id - Block id
+ * @param {string} filter.generatorPublicKey - Public key of delegate who generates the block
+ * @param {number} filter.numberOfTransactions - Number of transactions
+ * @param {string} filter.previousBlock - Previous block ID
+ * @param {number} filter.height - Block height
+ * @param {number} filter.totalAmount - Total amount of block's transactions
+ * @param {number} filter.totalFee - Block total fees
+ * @param {number} filter.reward - Block reward
+ * @param {number} filter.limit - Limit of blocks to retrieve, default: 100, max: 100
+ * @param {number} filter.offset - Offset from where to start
+ * @param {string} filter.sort - Sort order, default: height:desc
+ * @param {function} cb - Callback function
+ * @returns {function} cb - Callback function from params (through setImmediate)
+ * @returns {Object} cb.err - Error if occurred
+ * @returns {Object} cb.data - List of normalized blocks
+ */
+function _list(filter, cb) {
```

Comment:
Why do you name it with underscore at the beginning? 

In any case, I'd move to the top right after `BlocksController` function.

Reply:
I used the underscore because it is a "private" method.
But I'm refactoring this as requested by @MaciejBaj already ;)